x-pack/plugins/enterprise_search/public/applications/app_search/components/layout/kibana_header_actions.tsx

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+
+import { EuiButtonEmpty, EuiText, EuiFlexGroup, EuiFlexItem, EuiHeaderLinks } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+
+export const KibanaHeaderActions: React.FC = () => {
+  return (
+    <EuiHeaderLinks>
+      <EuiFlexGroup gutterSize="s">
```

Comment:
🤔 Super interesting, I hadn't come across EuiHeaderLinks before this! Do we really need it _and_ the nested EuiFlexGroup? I guess I'm just curious what EuiHeaderLinks is doing/adding either layout or style wise exactly. Or are you just using it as a semantic notation?

Reply:
From a quick QA test it looks like you can actually lose the `EuiFlexGroup` and only have `EuiHeaderLinks` as a wrapper...

<img width="500" alt="" src="https://user-images.githubusercontent.com/549407/119581936-774a1100-bd78-11eb-90c1-a5744bb53648.png">

Actually - wait. Looking at this DOM markup, we empathetically do **not** want to be using EuiHeaderLinks here. It has a very specific `nav` and aria label meaning which does not apply to our application. These are not navigation links, their application actions, and as such we definitely do not want to be using `EuiHeaderLinks` here. I'd drop that wrapper and keep `EuiFlexGroup`.

Reply:
Oh wow, good call! I just copied what WS was doing. I'll test this and if it all looks well for us I'll update theirs and ping them.

Reply:
Hey Constance, this stuff appears to be related to responsive behavior.

`EuiHeaderLinks` gives us this nice mobile behavior where it tucks everything into a menu.

The flex layout stacks elements on top of one another.

<img width="494" alt="a thing" src="https://user-images.githubusercontent.com/1427475/119718496-a5d9f180-be35-11eb-9a7b-fab9d16e5dab.png">

Reply:
Ah, that makes more sense to know it came from WS. It makes sense from them because (IIRC) their header actions actually _are_ links, and additionally are a separate form of navigation (to eweb and to the personal dashboard). That's not the case for us (query tester is a button, not a link), so we shouldn't copy them in this regard. Additionally, since we only have one action currently, we don't need the same mobile behavior as them. If this ever changes we can revisit that.